### ISSUE TITLE
wsgi:355 500 error for /journal/ and /account/*

### ISSUE THREAD

**Issue by alexjj** [aumayr/fava#147]
However the error is slightly different:

For /journal/:
It loads the page but just says: `Loading journal entries…`, but the terminal shows `E wsgi:355 500 GET /journal/`.

For any /account/ page it shows 

```
Internal Server Error

The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

and same terminal error. Once or twice it has loaded the assets page but mostly it doesn't.

Installed from pip3.

**Comment by corani** [comment]
Could you start the program with `--debug` and post the full stacktrace
here?

**Comment by alexjj** [comment]
Both errors caused by `TypeError: Decimal('1.5080') is not JSON serializable`

```
File "/usr/lib/python3.5/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.5/site-packages/fava/application.py", line 111, in journal
    'data': app.api.journal(with_change_and_balance=app.config.user.getboolean('journal-general-show-balances'))
  File "/usr/lib/python3.5/site-packages/flask/json.py", line 238, in jsonify
    indent=indent),
  File "/usr/lib/python3.5/site-packages/flask/json.py", line 126, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/usr/lib/python3.5/json/__init__.py", line 237, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.5/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.5/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.5/site-packages/fava/api/serialization.py", line 26, in default
    return JSONEncoder.default(self, o)
  File "/usr/lib/python3.5/site-packages/flask/json.py", line 83, in default
    return _json.JSONEncoder.default(self, o)
  File "/usr/lib/python3.5/json/encoder.py", line 180, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: Decimal('1.5080') is not JSON serializable
```

**Comment by aumayr** [comment]
Thanks @alexjj for the bug report!

Can you try to update fava to the newest version? (Did you install via `pip` or from source?)

Judging from this line it seems that your version is outdated: 

>  File "/usr/lib/python3.5/site-packages/fava/application.py", line 111, in journal
>     'data': app.api.journal(with_change_and_balance=app.config.user.getboolean('journal-general-show-> balances'))

Compared to the current version: 

https://github.com/aumayr/fava/blob/master/fava/application.py#L117-L119

**Comment by alexjj** [comment]
I've installed from pip3 and it says it's the latest version, beancount-fava-0.2.3. I tried upgrading/uninstalling, reinstalling and still the same version/error.

I can build from source later if that's all that's needed to fix it. Thanks.

**Comment by aumayr** [comment]
Please let me know if installing from source did fix the problem!

**Comment by alexjj** [comment]
Built from source but now...

```python
Traceback (most recent call last):
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/alex/dev/finance/fava/fava/application.py", line 119, in journal
    return render_template('journal.html')
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/templating.py", line 128, in render_template
    context, ctx.app)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/flask/templating.py", line 110, in _render
    rv = template.render(context)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/alex/.virtualenvs/fava/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/alex/dev/finance/fava/fava/templates/journal.html", line 2, in top-level template code
    {% set active_page = 'journal' %}
  File "/home/alex/dev/finance/fava/fava/templates/_layout.html", line 105, in top-level template code
    {% block content %}{% endblock %}
  File "/home/alex/dev/finance/fava/fava/templates/journal.html", line 6, in block "content"
    {% with journal = api.journal(), show_tablefilter=True, show_change_and_balance=config.user.getboolean('journal-general-show-balances')  %}
  File "/home/alex/dev/finance/fava/fava/api/__init__.py", line 317, in journal
    return self._journal(self.entries, with_change_and_balance=with_change_and_balance)
  File "/home/alex/dev/finance/fava/fava/api/__init__.py", line 178, in _journal
    entry = serialize_entry(posting)
  File "/home/alex/dev/finance/fava/fava/api/serialization.py", line 56, in serialize_entry
    new_entry['balance'] = entry.diff_amount + entry.amount
TypeError: unsupported operand type(s) for +: 'Amount' and 'Amount'
```

**Comment by yagebu** [comment]
Good catch! Can reproduce and fixed it in my PR.

**Comment by aumayr** [comment]
The error initially reported was also addressed in 10cf818
